contribs/src/main/java/com/netflix/conductor/contribs/http/HttpTask.java

```diff
@@ -54,13 +55,75 @@
 /**
  * @author Viren
  * Task that enables calling another http endpoint as part of its execution
+ * 
+ * Basic support is to fail responses < 200 and > 299
+ * If you desire additional support for any specific response code  
+ * set map input_Parameter:status_support, keys are httpStatuses and value is TaskStatus to be set
+ * Use a decision tree task after completion to handle specific httpStatuses
+ * input_Parameter:alternate_workflow can be used to indicated default workflow for httpStatuses that are invoked.
  */
+/**
+ * @author jstanley 
+ * 
+ * Task updateds to allow http statuses (>200 and > 299)
+ * to be handled with custom handlers.
+ * 
+ * Basic support remains the same to fail responses < 200 and > 299 (exactly the same
+ * as orginally designed) Setting http_status_override values overrides default
+ * failure for statuses outside the range.
+ * 
+ * If you desire additional support for any specific response code set
+ * map input_parameter:status_support, keys are httpStatuses and value
+ * is TaskStatus to be set. Use a decision tree task after completion to
+ * handle specific httpStatuses.
+ * 
+ * outputParameters now include httpStatus and overrideActivated where overrideActivated
+ * can be used to 
+ * 
+ * Example: erp task has type HTTP and sets http_status_support
+ *          to 404:COMPLETE, 409:COMPLETE. If 404 or 409 is the returned http_status then
+ *          overrideActivated is set to true. 
+ *          A DECISION Task can be used to handle each case with a 
+ *          unique following task/workflow. 
+ *          OR
+ *          Use the overrideActivated parameter to have all returned error conditions to
+ *          use the same task/workflow.
+ * 
+ * Example code:
+ * [
+ * {
+ *  "name": "http",
+ *  "taskReferenceName": "erp",
+ *     "type": "HTTP",
+ *     "inputParameters": {
+ *       "http_request": {
+ *         "method": "GET",
+ *         "accept": "application/json",
+ *         "contentType": aplication/json",
+ *         "headers": {
+ *           "Authorization": {
+ *           }
+ *         },
+ *         "uri": {
+ *           
+ *         }
+ *       },
+ *       "http_status_override": {
+ *         "404": "COMPLETED",
+ *         "409": "COMPLETED"
+ *       }
+ *     }
+ *   }
+ * ]
+*/
 @Singleton
 public class HttpTask extends WorkflowSystemTask {
```

Comment:
So you're modifying their class - would it be better to create a new subclass that extends theirs and adds this additional support?  I'm not suggesting that - just wondering your thoughts.

Reply:
Yeah, I would normally but unfortunately the design is such that a significant portion of the code would have to be pulled because the validation is in the "main" method. Its a reasonable choice and it would guarantee that functionality would not change from the original. Funny, that was actually my original design. Simple to redo it.  It would also require duplication of the Unit tests I believe since most of the testing is concerned with testing the start functionality.

Reply:
I think sub-classing would be ideal.  Having said that, if the logic changes here don't change the behavior of the class when the status overrides are not used,  changing the class itself is okay.